### PR TITLE
preproc: varargscount: bugfix for tokens starting with '(' not being counted

### PR DESCRIPTION
**Pull request by mwierzbix**
; also introduced unit tests to check the issue

Signed-off-by: Michal Jerzy Wierzbicki <michalx.wierzbicki@linux.intel.com>

varargscount is macro META_COUNT_VARARGS_BEFORE_COMPILE

Fix for issue https://github.com/thesofproject/sof/issues/542

### CODE REVIEW

File: test/cmocka/src/debugability/macros.c [hunk 1]
```diff
@@ -45,15 +45,20 @@
...
+	));
 	const char *should_be_eq =
 		"__attribute__((section(\".static_log.\""
 		" \"LOG_LEVEL_CRITICAL\"))) "
 		"static const struct { uint32_t level; uint32_t component_id; "
 		"uint32_t params_num; uint32_t line_idx; uint32_t file_name_len; "
 		"const char file_name[sizeof(\"src/debugability/macros.c\")]; "
 		"uint32_t text_len; const char text[sizeof(\"Message\")]; } "
-		"log_entry = { 1";
+		"log_entry = { 1(6 << 24)152sizeof(\"src/debugability/macros.c\")"
```
Comment (mwierzbix) [inline test/cmocka/src/debugability/macros.c]: I am not sure about this one(that `)152sizeof` seems sketchy), I made the change just to make unit tests shut up.
As @akloniex  pointed out to me this bug could be severe so in the scope of this PR I'd like to let this slide,
but I'll also want to sit down with @akloniex on what should be the correct value here as he will be modifying this very const char *should_be_eq here for https://github.com/thesofproject/sof/pull/530.

File: test/cmocka/src/debugability/macros.c [hunk 2]
```diff
@@ -78,32 +83,29 @@ static char *get_should_be(const int param_count)
...
+	/*4*/") ? 1 : -1]; log_func log_function = (log_func)& _trace_event",
+	/*5*/param_count,
+	/*6*/"; log_function(&log_entry",
+	/*7*/maybe_comma,
+	/*8*/paramlist,
+	/*9*/");}"
 	);
 	if (paramlist)
 		free(paramlist);
+	// TODO: maybe remove all whitespace chars; they're not important here
```
Comment (mwierzbix) [inline test/cmocka/src/debugability/macros.c]: I am thinking about removing all whitespaces for this check in the future, as it really doesn't matter whether preprocessor gives us
`uint32_t param0 , uint32_t...` or
`uint32_t param0,uint32_t...`
as the compiler doesn't care.

Again, I'd rather not do this in bugfix PR.

File: test/cmocka/src/lib/preproc/varargs_count.c [hunk 3]
```diff
@@ -51,35 +51,49 @@ static void META_CONCAT_SEQ_DELIM_(prefix, test_func, postfix)(void **state)\
...
-TEST_HERE_DECLARE(META_NAME, 1, 1, A      )
-TEST_HERE_DECLARE(META_NAME, 3, 3, A, B, C)
-TEST_HERE_DECLARE(PP_NARG  , 0, 0,        )
-TEST_HERE_DECLARE(PP_NARG  , 1, 1, A      )
-TEST_HERE_DECLARE(PP_NARG  , 3, 3, A, B, C)
+#define TEST_HERE_DECLARE_GROUP(func_name)\
+	TEST_HERE_DECLARE(func_name, 0, 0,        )\
+	TEST_HERE_DECLARE(func_name, 1, 1, A      )\
+	TEST_HERE_DECLARE(func_name, 3, 3, A, B, C)\
+	TEST_HERE_DECLARE(func_name, with_parenthesis_pre,  1,\
```
Comment (mwierzbix) [inline test/cmocka/src/lib/preproc/varargs_count.c]: New unit tests.

**Comment by mwierzbix** [comment]
Great, Travis build for SUE hanged during `docker pull` (it is, even before attempting to build).
https://travis-ci.org/thesofproject/sof/jobs/451169579
This day cannot get any better it seems.

**Comment by xiulipan** [comment]
@mwierzbix 
Do not worry, I will retrigger the build for you.
And I think Travis have some magic word to rebuild the tset.

> If you've sent a pull request: You can close the PR then open it again. This will trigger a new build.
https://stackoverflow.com/questions/17606874/trigger-a-travis-ci-rebuild-without-pushing-a-commit

File: test/cmocka/src/debugability/macros.c (hunk 1 above)
Comment (akloniex) [reply test/cmocka/src/debugability/macros.c]: @mwierzbix It seems like CAPTURE macro improperly removes commas from input. We'll look into that.

**Comment by lgirdwood** [comment]
@mwierzbix Fwiw, Travis can be restarted if you click on details and the rebuild.

**Comment by mwierzbix** [comment]
@lgirdwood @xiulipan Thanks :)
Adding to what you said, you also have to be logged into Travis with your Github account (probably one being contributor to _thesofproject_) in order to schedule restart.

File: test/cmocka/src/debugability/macros.c (hunk 1 above)
Comment (mwierzbix) [reply test/cmocka/src/debugability/macros.c]: Yeah, it's in fact the `CAPTURE` macro getting rid of `,`s. So it's a non-issue here.

**Comment by lgirdwood** [comment]
Added @slawblauciak for cmocka parts

**Pull request merged**